cryodrgn/source.py

```diff
@@ -0,0 +1,278 @@
+import os.path
+from collections.abc import Iterable
+from concurrent import futures
+import numpy as np
+import pandas as pd
+from typing import Union, List
+from cryodrgn.starfile import Starfile
+from cryodrgn.mrc import MRCHeader
+
+
+class ImageSource:
```

Comment:
Should this be called `ImageDataset` and be a subclass of `torch.Dataset[np.ndarray]`? The way these datasets seem to be used is always to be fed to a `DataLoader` which expects a `Dataset`, no?

Docstring would be nice here explaining that this stores an array of images loaded from some data source, and which may be eagerly or lazily loaded.